src/Nat.mo

```diff
@@ -1,72 +1,299 @@
+  /// value to pass to a higher order function. It is not possible to use `<`
+  /// as a function value at the moment.
   public func less(x : Nat, y : Nat) : Bool { x < y };
 
-  /// Returns `x <= y`.
+  /// "Less than or equal" function for Nat types.
+  /// This is equivalent to `x <= y`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// ignore Nat.lessOrEqual(1, 2); // => true
+  /// 1 <= 2 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `<=` operator) is so that you can use it as a function
+  /// value to pass to a higher order function. It is not possible to use `<=`
+  /// as a function value at the moment.
   public func lessOrEqual(x : Nat, y : Nat) : Bool { x <= y };
 
-  /// Returns `x > y`.
+  /// "Greater than" function for Nat types.
+  /// This is equivalent to `x > y`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// ignore Nat.greater(2, 1); // => true
+  /// 2 > 1 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `>` operator) is so that you can use it as a function
+  /// value to pass to a higher order function. It is not possible to use `>`
+  /// as a function value at the moment.
   public func greater(x : Nat, y : Nat) : Bool { x > y };
 
-  /// Returns `x >= y`.
+  /// "Greater than or equal" function for Nat types.
+  /// This is equivalent to `x >= y`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// ignore Nat.greaterOrEqual(2, 1); // => true
+  /// 2 >= 1 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `>=` operator) is so that you can use it as a function
+  /// value to pass to a higher order function. It is not possible to use `>=`
+  /// as a function value at the moment.
   public func greaterOrEqual(x : Nat, y : Nat) : Bool { x >= y };
 
-  /// Returns the order of `x` and `y`.
+  /// General purpose comparison function for `Nat`. Returns the `Order` (
+  /// either `#less`, `#equal`, or `#greater`) of comparing `x` with `y`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// Nat.compare(2, 3) // => #less
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `>=` operator) is so that you can use it as a function
+  /// value to pass to a higher order function. It is not possible to use `>=`
+  /// as a function value at the moment.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// import Array "mo:base/Array";
+  /// Array.sort([2, 3, 1], Nat.compare) // => [1, 2, 3]
+  /// ```
   public func compare(x : Nat, y : Nat) : { #less; #equal; #greater } {
     if (x < y) { #less } else if (x == y) { #equal } else { #greater }
   };
 
   /// Returns the sum of `x` and `y`, `x + y`.
+  ///
+  /// Example:
```

Comment:
Maybe it could help to mention (again): No overflow since `Nat` has infinite precision.